blog-app/client/js/script.js

```diff
@@ -0,0 +1,590 @@
+let articles;
+
+window.onload = function(e) {
+    if(window.location.hash === ""){
+        window.location.hash = "home";
+    }
+    onRouteChange(e);
+    window.scrollTo(0, 0);
+    // set the theme
+    let localStorageTheme = localStorage.getItem('theme');
+    if(localStorageTheme === 'dark'){
+        document.documentElement.setAttribute('data-theme', 'dark');
+    } else {
+        document.documentElement.setAttribute('data-theme', 'light');
+    }
+    
+}
+
+window.addEventListener('hashchange', e => this.onRouteChange(e));
+
+function onRouteChange(e) {
+    window.scrollTo(0, 0);
+    const hashLocation = window.location.hash.substring(1);
+    this.loadContent(hashLocation);
+}
+
+// call getArticles to load the content
+function loadContent(uri){
+    getArticles(uri);  
+}
+
+//FUNCTIONS
+// get articles from db.json
+function getArticles(uri) {
+
+    fetch('http://localhost:3000/articles')
+    .then(
+        function(response) {
+        if (response.status !== 200) {
+            console.log('Looks like there was a problem. Status Code: ' +
+            response.status);
+            return;
+        }
+
+        // Examine the text in the response
+        response.json().then(function(data) {
+            articles = data;
+            // call the displayHtml function to create the elements for the page based on the uri
+            displayHtml(uri);
+            if(uri === "home") {
+
+                // CONSTANTS
+                const ADDBTN = document.getElementById("add-article");
```

Comment:
you could move the code for the modal inside the displayHtml function, where you check for the path "home"

Reply:
I updated the code.